src/+datools/+utils/+stat/RH.m

```diff
@@ -10,25 +10,46 @@ function RH(model, xt)
     fprintf('need to pass the truth at this time step');
     exit;
 end
-var = length(model.RankHistogram);
+var = length(filter.RankHistogram);
+varobservtion = observation.Indices;
+xa = filter.Ensemble;
+[~, ensN] = size(xa);
 
-xf = model.Ensemble;
-[~, ensN] = size(xf);
-
-for i = 1:var
-    tempVar = model.RankHistogram(i);
-    model.RankValue(i, ensN+2) = tempVar;
-    ensFor = sort(xf(tempVar, :));
-
-    if xt(tempVar) < ensFor(1)
-        model.RankValue(i, 1) = model.RankValue(i, 1) + 1;
-    elseif xt(tempVar) > ensFor(length(ensFor))
-        model.RankValue(i, ensN+1) = model.RankValue(i, ensN+1) + 1;
-    else
-        index = find(ensFor > xt(tempVar), 1);
-        model.RankValue(i, index) = model.RankValue(i, index) + 1;
-    end
 
+switch observationvariable
+    case 'Truth'
```

Comment:
can we not have double cases? There should be one procedure to compute the rank histogram(ensemble, reference). the reference input to this takes in either truth or observation, that should be done elsewhere. The idea with software is to not have repetitive code like this.